.github/workflows/gradle-check.yml

```
@@ -50,42 +70,44 @@ jobs:
           echo "pr_number=Null" >> $GITHUB_ENV
 
       - name: Checkout opensearch-build repo
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
         uses: actions/checkout@v4
         with:
           repository: opensearch-project/opensearch-build
           ref: main
           path: opensearch-build
 
       - name: Trigger jenkins workflow to run gradle check
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
         run: |
           set -e
           set -o pipefail
           bash opensearch-build/scripts/gradle/gradle-check.sh ${{ secrets.JENKINS_GRADLE_CHECK_GENERIC_WEBHOOK_TOKEN }} | tee -a gradle-check.log
 
       - name: Setup Result Status
-        if: always()
+        if: always() && steps.changed-files-specific.outputs.any_changed == 'true'
```

I think `always()` does not apply anymore:

```suggestion
        if: steps.changed-files-specific.outputs.any_changed == 'true'
```

---

If `always()` is removed, the step would be skipped if previous check failed as default is `success()`

---

`failure() && steps.changed-files-specific.outputs.any_changed == 'true'` will be evaluated to false if previous step fails right? In that case we need always() for first condition to be set to true

---

You might be right, thanks @gaiksaya , it seems like semantics of `always()` is more than just `true`